File: rev-com-server/src/main/java/com/codergb/bean/book/Book.java
package com.codergb.bean.book;

import com.codergb.bean.FileInfo;
import lombok.*;

import java.util.Date;

@Data
@AllArgsConstructor
@NoArgsConstructor
@ToString
@EqualsAndHashCode(callSuper = true)
public class Book extends FileInfo {
  private String id;
  private String name;
  private Writer writer;
  private Publish publish;
  private Date publishTime;
  private String coverUrl;
  private Double price;
  private String description;
  private Integer pageCount;
  private Double score;
}

File: rev-com-manage/src/types/book/book.ts
import { IWriter } from "@/types/book/writer"
import { IPublish } from "@/types/book/publish"

interface IBook {
  id: string
  name: string
  writer: IWriter
  publish: IPublish
  publishTime: string
  coverUrl: string
  price: number
  description: string
  pageCount: number
  score: number
}
export type { IBook }
